### PR TITLE
add extra method for use by unregister RPC api

### PR DESCRIPTION
**Pull request by arnej27959**
I confirm that this contribution is made under the terms of the license found in the root directory of this repository's source tree and that I have the authority necessary to make this contribution on behalf of its copyright owner.

@havardpe please review

### CODE REVIEW

File: slobrok/src/vespa/slobrok/server/local_rpc_monitor_map.h [hunk 1]
```diff
@@ -126,6 +129,9 @@ class LocalRpcMonitorMap : public MapListener,
     void addLocal(const ServiceMapping &mapping,
                   std::unique_ptr<ScriptCommand> inflight);
 
+    /** for use by unregister API, returns true on success */
+    bool removeLocal(const ServiceMapping &mapping);
```
Comment (havardpe) [inline slobrok/src/vespa/slobrok/server/local_rpc_monitor_map.h:133]: consider using wouldConflict in the consensus map instead to get this bool (is this a 'clean' remove)

**Review by havardpe — COMMENTED** [review]
might need to clear ignoreFirstOk when reusing the per service object.

**Comment by arnej27959** [comment]
@havardpe please take another look

File: slobrok/src/vespa/slobrok/server/local_rpc_monitor_map.cpp [hunk 2]
```diff
@@ -93,6 +93,7 @@ void LocalRpcMonitorMap::addLocal(const ServiceMapping &mapping,
         if (exists.spec == mapping.spec) {
             LOG(debug, "added mapping %s->%s was already present",
                 mapping.name.c_str(), mapping.spec.c_str());
+            // clear exists.ignoreFirstOk ?
```
Comment (havardpe) [inline slobrok/src/vespa/slobrok/server/local_rpc_monitor_map.cpp:96]: another question; should we really return 'ok' when we known the service is in the process of being removed?

**Review by havardpe — APPROVED** [review]

**Pull request merged**